pkg/api/validation/validation.go

```diff
@@ -1022,11 +1022,19 @@ func ValidatePersistentVolumeClaim(pvc *api.PersistentVolumeClaim) field.ErrorLi
 func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *api.PersistentVolumeClaim) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newPvc.ObjectMeta, &oldPvc.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidatePersistentVolumeClaim(newPvc)...)
-	// if a pvc had a bound volume, we should not allow updates to resources or access modes
-	if len(oldPvc.Spec.VolumeName) != 0 {
+	// PVController needs to update PVC.Spec w/ VolumeName. Any other change is not allowed.
+	// Claims are immutable in order to enforce quota, range limits, etc. without gaming the system.
+	if len(oldPvc.Spec.VolumeName) == 0 {
+		// Only volumeName changes are allowed. Reset back to empty string after equality check
+		oldPvc.Spec.VolumeName = newPvc.Spec.VolumeName
 		if !api.Semantic.DeepEqual(newPvc.Spec, oldPvc.Spec) {
-			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "spec is immutable once a claim has been bound to a volume"))
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "spec is immutable after creation"))
 		}
+		oldPvc.Spec.VolumeName = ""
+	}
+	// after a pvc is bound to a volume, no changes are allowed
+	if len(oldPvc.Spec.VolumeName) != 0 {
```

Comment:
Also, there are no updates to PVC.Spec in binding other than volumeName.  The _actuals_ for a user are on PVC.Status.